specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./cdn.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Policy": {
+            "$ref": "./examples/WafPolicyGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "WebApplicationFirewallPolicies"
+        ],
+        "description": "Create or update policy with specified rule set name within a resource group.",
+        "operationId": "Policies_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PolicyNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "description": "Policy to be created.",
+            "in": "body",
+            "name": "cdnWebApplicationFirewallPolicy",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new protection policy has been created.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./cdn.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates specific policy": {
+            "$ref": "./examples/WafPolicyCreateOrUpdate.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "patch": {
```

Comment:
While PATCH is only mandatory for tags, we prefer to see PATCH support for all modifiable properties.

Reply:
@sushantsingh any response to this?

Reply:
We will do this in the stable version of the resource. As of now, as recommended in the Arm API review meeting, the minimum bar is to be able to update tags and that has been included in the API.
We do not have any scenario where we update parts of the config, the entire config is updated as a single entity.